python/CMakeLists.txt

```diff
@@ -499,6 +500,8 @@ if(PYARROW_BUILD_FLIGHT)
                        ${ARROW_SO_VERSION})
       if(MSVC)
         bundle_arrow_implib(ARROW_FLIGHT_SHARED_IMP_LIB)
+        bundle_arrow_dependency(cares)
+        bundle_arrow_dependency(libprotobuf)
```

Comment:
We may also need to bundle `libcrypto` and `libssl`? @xhochy what do you think?

Reply:
Might need to. What does `pip install grpcio` do? We might consider statically linking boringssl in the wheel if that's what Google is doing

Reply:
grpcio seems to statically link whatever variant of OpenSSL they chose. I'm not sure that's a good idea (we need to follow security releases of OpenSSL).

Reply:
They are probably using BoringSSL then so they are in unilateral control of what's used

For the record, I just built a wheel off master that links against OpenSSL 1.0.x ABI version and tried using it in a conda environment with OpenSSL 1.1.x and it fails to link

```
$ python
Python 3.7.3 | packaged by conda-forge | (default, Mar 27 2019, 23:01:00) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyarrow
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/wesm/miniconda/envs/turbodbc-dev/lib/python3.7/site-packages/pyarrow/__init__.py", line 49, in <module>
    from pyarrow.lib import cpu_count, set_cpu_count
ImportError: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory
```

So it seems that not only should we bundle OpenSSL with the wheel, we should also probably upgrade OpenSSL in the manylinux1 image to 1.1.x

https://issues.apache.org/jira/browse/ARROW-5642

Reply:
Since we need to bundle OpenSSL, perhaps we might just as well link statically after all (the bundling code on the Python side is a PITA).